File: swift/write_to_named_pipe.swift
import Foundation

let pipePath = "output.pipe"

if !FileManager.default.fileExists(atPath: pipePath) {
    mkfifo(pipePath, 0o644)
}

if let pipe = fopen(pipePath, "w") {
    fputs("Hello World!\n", pipe)
    fclose(pipe)
}

File: php/write_to_named_pipe.php
<?php

$pipePath = 'output.pipe';

if (!file_exists($pipePath)) {
    posix_mkfifo($pipePath, 0666);
}

$pipe = fopen($pipePath, 'w') or die("Could not open '$pipePath'");

fwrite($pipe, "Hello World!\n");

fclose($pipe);
